src/instructlab/model/serve_backend.py

```diff
@@ -23,6 +25,38 @@ def signal_handler(num_signal, __):
     sys.exit(0)
 
 
+def get_gpu_count() -> int:
+    # Third Party
+    import torch
+
+    if torch.cuda.is_available():
+        return torch.cuda.device_count()
+    return 0
+
+
+def check_tensor_parallel_size(vllm_args: List) -> Tuple[bool, int, int]:
+    if contains_argument("--tensor-parallel-size", vllm_args):
+        # First Party
+        from instructlab.model.backends.vllm import get_argument
+
+        available_gpus = get_gpu_count()
+        effective_gpus = 0
+        try:
+            tps = get_argument("--tensor-parallel-size", vllm_args)
+            if tps is not None and tps is not True:
+                effective_gpus = int(tps)
+        except ValueError:
+            return (False, 0, 0)
+        # if gpu is 0, --tensor-parallel-size is 1 that should be normal
```

Comment:
Not following this statement.  Is gpu here referring to the --gpus arg?

Reply:
sorry, wrong statement, will remove it.